### PR TITLE
Use Ethereum.selectedAddress as the source of truth

### PR DESCRIPTION
**Pull request by MelvinMallari**
Tests are failing, telling me to use `this.$store` for `getModule()`.. but I am? Jest is acting really weird on this one, thoughts?

### CODE REVIEW

**Comment by MelvinMallari** [comment]
PR for tickets #89 #90

File: tests/unit/ui/Navigation.spec.ts [hunk 1]
```diff
@@ -39,8 +38,10 @@ describe('FlashMessage.vue', () => {
...
-    expect(wrapper.findAll(`.${navbarClass} .${navbarMenuClass} .${navbarEndClass} .${tileClass}`).length).toBe(1)
-    expect(wrapper.findAll(`.${navbarClass} .${navbarMenuClass} .${navbarEndClass} .${connectClass}`).length).toBe(1)
-    expect(wrapper.findAll(`.${navbarClass} .${navbarMenuClass} .${navbarEndClass} .${tileClass} .${logoClass}`).length).toBe(1)
+    expect(wrapper.findAll(`.${navbarClass} .${navbarMenuClass} .${navbarEndClass}`).length).toBe(1)
+    expect(wrapper.find(`.${connectClass}`).isVisible()).toBe(true)
+    expect(wrapper.findAll(`.${connectClass}`).length).toBe(1)
+    expect(wrapper.find(`.${tileClass}`).isVisible()).toBe(false)
+    expect(wrapper.findAll(`.${tileClass}`).length).toBe(1)
   })
 })
```
Comment (thirtyaughtsix) [inline tests/unit/ui/Navigation.spec.ts]: We should have specs for the new behaviors.

File: src/modules/UploadModule.ts [hunk 2]
```diff
@@ -102,11 +105,12 @@ export default class UploadModule extends VuexModule implements FfaProcessModule
...
 
   get hash(): string {
-    const metaMaskModule = this.context.rootState.metaMaskModule as MetaMaskModule
     const web3Module = this.context.rootState.web3Module as Web3Module
+    const metaMaskModule = this.context.rootState.metaMaskModule as MetaMaskModule
+    const flashesModule = this.context.rootState.web3Module as FlashesModule
 
-    if (metaMaskModule.publicKey.length === 0) {
-      throw new Error(Errors.PUBLIC_KEY_EMPTY)
+    if (typeof ethereum === 'undefined' || typeof ethereum.selectedAddress === 'undefined') {
```
Comment (thirtyaughtsix) [inline src/modules/UploadModule.ts]: This isn't working.  Connect metamask and then reload the page.  Metamask is disconnected.

This also violates the same engineering principle (separation of concerns) twice:

1. This method is a getter.  It's role is to define a computed property.  That's all it should ever do.

2. This class is a vuex module.  It should store and retrieve state.  Business logic doesn't belong here.

This block needs to be moved elsewhere where business logic is allowed like `App.vue`.

The logic needs specs (see comment in `Navigation.spec.ts`), the actual connect stuff in a new `App.spec.ts` and the changes in the rendering of the nav in `Navigation.spec.ts`.

File: src/views/List.vue [hunk 3]
```diff
@@ -52,6 +53,20 @@ export default class List extends Vue {
   protected dropzoneRef = 'dropzone'
   protected dropzone!: Dropzone
 
+  private mounted() {
+    const flashesModule = getModule(FlashesModule, this.$store)
+    const metaMaskModule = getModule(MetaMaskModule, this.$store)
+    const web3Module = getModule(Web3Module, this.$store)
+
+    if (typeof ethereum === 'undefined' || typeof ethereum.selectedAddress === 'undefined') {
```
Comment (thirtyaughtsix) [inline src/views/List.vue]: Having the same exact code twice is serious code smell and indicates the logic isn't correct.  Remove this and rely on a single correct block as per the comment above.

File: src/views/List.vue [hunk 4]
```diff
@@ -52,6 +53,20 @@ export default class List extends Vue {
...
+      setPublicKey(flashesModule, metaMaskModule, web3Module)
+    }
+
+    if (typeof web3Module.web3.eth === 'undefined') {
```
Comment (thirtyaughtsix) [inline src/views/List.vue]: This should also be moved to `App.vue`.

**Comment by thirtyaughtsix** [comment]
@MelvinMallari are we closing ths PR?  It's superseded by #111 right?

**Pull request closed without merging**